eunkyung/services/userService.js

```diff
@@ -0,0 +1,37 @@
+const userDao = require("../models/userDao");
+const bcrypt = require("bcrypt");
+const jwt = require("jsonwebtoken");
+
+const createUser = async (name, password, email, profileImage) => {
+  const saltRounds = 12;
+  const hashedPw = await bcrypt.hash(password, saltRounds);
+
+  const createUserResult = await userDao.createUser(
+    name,
+    hashedPw,
+    email,
+    profileImage
+  );
+  return createUserResult;
+};
+
+const emailCheckLogin = async (email, password) => {
+  const [userInfo] = await userDao.emailCheckLogin(email);
+  if (!userInfo) {
+    return undefined;
+  }
+  const checkPw = await bcrypt.compare(password, userInfo.password);
+
+  if (checkPw === true) {
+    const payLoad = { exp: 60000 };
+    const secretKey = "mySecretKey";
+    return jwt.sign(payLoad, secretKey);
+  } else {
+    return "wrongPw";
+  }
```

Comment:
1. checkPw에 대한 결과가 true라면 jwt 토큰을 발급한다.
2. checkPw에 대한 결과가 false라면 유저 정보가 틀렸다는 에러를 반환한다.

- checkPw 자체는 true와 false를 반환하기 합니다.
그렇기 때문에 checkPw가 true라면 조건문을 작성하지 않아도 그 아래의 로직이 실행이 될 것입니다.
checkPw가 false를 반환할 때만 예외처리를 해주면 될 것 같습니다.
- secretKey는 해당 토큰이, 우리 서버가 발행한 토큰임을 확인할 수 있는 정보이므로 환경변수로 관리해주세요.
- 토큰의 payload의 클레임으로 user 정보를 특정할 수 있지만, 민감한 개인정보가 포함되지 않도록 수정해주세요!

```suggestion
  const payLoad = { userId: userInfo.id };
  const secretKey = process.env.secretKey;

  if ( !checkPw ) {
    throw new Error("Invalid Password");
  }
  
  return jwt.sign(payLoad, secretKey);
```